### ISSUE TITLE
retain_mut is deprecated

### ISSUE THREAD

**Issue by paolobarbolini** [rustsec/advisory-db#1929]
As described in the README, `retain_mut` has been deprecated with the release of Rust 1.61, which stabilized [`Vec::retain_mut`](https://doc.rust-lang.org/nightly/std/vec/struct.Vec.html#method.retain_mut) and [`VecDeque::retain_mut`](https://doc.rust-lang.org/nightly/std/collections/struct.VecDeque.html#method.retain_mut).

I see this crate still makes peaks of 30k downloads a day, with Rust 1.61 having been released almost 2 years ago. Should an advisory be made for it?

**Comment by paolobarbolini** [comment]
Cc @upsuper

**Comment by 8573** [comment]
The crate's README says:
> This crate has been deprecated. Rust 1.61 stabilized retain_mut for Vec and VecDeque, so you can use them directly. This crate is no longer maintained.

I don't think RustSec usually carries mere deprecation warnings, but the last sentence is [more relevant](https://github.com/rustsec/advisory-db/blob/799ff4a10673405b2334f6653519fb092aa99845/HOWTO_UNMAINTAINED.md).

**Comment by upsuper** [comment]
While I'm not sure how useful such an advisory would be, I'm not against issuing one. It is indeed unmaintained, but on the other hand, it is also very unlikely keeping the current implementation would have any security implication.

**Comment by jayvdb** [comment]
There are quite a few unsafe in this crate.  As the repo is archived, people cant raise issues if they are found.  It is unmaintained.  Users should be advised.

**Comment by djc** [comment]
I agree it would make sense to publish an unmaintained advisory for this, and it looks like we have agreement from the maintainer.
